src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -210,15 +210,19 @@ public SimpleInterval getCurrentReadShardBounds() {
      * Returns the default list of CommandLineReadFilters that are used for this tool. The filters
      * returned by this method are subject to selective enabling/disabling and customization by the
      * user via the command line. The default implementation uses the {@link WellformedReadFilter}
-     * filter with all default options. Subclasses can override to provide alternative filters.
+     * filter with all default options, as well as the {@link ReadFilterLibrary.MappedReadFilter}.
+     * Subclasses can override to provide alternative filters.
      *
      * Note: this method is called before command line parsing begins, and thus before a SAMFileHeader is
      * available through {link #getHeaderForReads}.
      *
      * @return List of default filter instances to be applied for this tool.
      */
     public List<ReadFilter> getDefaultReadFilters() {
-        return Collections.singletonList(new WellformedReadFilter());
+        final List<ReadFilter> defaultFilters = new ArrayList<>(2);
+        defaultFilters.add(new WellformedReadFilter());
+        defaultFilters.add(new ReadFilterLibrary.MappedReadFilter());
```

Comment:
No, `HaplotypeCaller` filters out all unmapped reads, and the traversal cannot handle them, so this filter should be on by default for AssemblyRegionTraversal.